connectors/src/connectors/microsoft/temporal/activities.ts

```diff
@@ -759,3 +764,97 @@ async function updateParentsField({
     parents,
   });
 }
+
+export async function microsoftDeletionActivity({
+  connectorId,
+  nodeIdsToDelete,
+}: {
+  connectorId: ModelId;
+  nodeIdsToDelete: string[];
+}): Promise<Result<void, Error>> {
+  const connector = await ConnectorResource.fetchById(connectorId);
+  if (!connector) {
+    return new Err(new Error(`Connector ${connectorId} not found`));
+  }
+
+  const dataSourceConfig = dataSourceConfigFromConnector(connector);
+
+  async function deleteNodeAndChildren(
```

Comment:
would move that to file.ts and make use of the existing `deleteFile` and `deleteFolder` there

Also I suggest calling  it `recursivelyDeleteNode`